elm-format-lib/src/Parse/String.hs

```diff
@@ -265,31 +257,30 @@ multiString pos end row col initialPos sr sc revChunks =
 -- ESCAPE CHARACTERS
 
 
-data Escape
-  = EscapeNormal
-  | EscapeUnicode !Int !Int
+data EscapeResult
+  = EscapeOk !Int ES.Chunk
```

Comment:
? might not have been necessary to combine these after all?

Reply:
This is adressed in 5a2d419.

#### Background.
TL;DR: I've changed my mind an now agree with Aarons comment, better to _not_ combine them after all.

The reason the Elm compiler has these two variants in the first place is because when encountering "normal escapes" (i.e `\[a-z]`), it can just include them as is in the JS output. Whereas `\u{....}` escapes needs to be transformed. In the original compiler code, `ES.Slice` is used for "normal escapes" and `ES.CodePoint` for `\u{....}` style.

https://github.com/avh4/elm-format/blob/778137761cd11f8a3010967d4fbd0ea8c945a0a1/elm-format-lib/src/Parse/String.hs#L79-L83

But in elm-format we normalize all escape sequences regardless so this distinction becomes less significant. The two branches are very similar now:

https://github.com/avh4/elm-format/blob/5a2d41983098bace966c082c638ff83668c87645/elm-format-lib/src/Parse/String.hs#L78-L82

An urge arises to generalize and have a single variant and branch for this which is indeed quite straightforward. But there is a value in having the code be similar to the compiler, which is why I agree with reverting to two variants again.